reactComponents/src/searchbar/tokenSearch/SearchInput.tsx

```diff
@@ -1,90 +1,97 @@
   margin-right: auto;
   position: relative;
-  border-color: #067c82;
-  border-style: solid;
-  border-width: 2px;
-  border-radius: 30px;
-  background: #08333c;
-  padding: 11px 15px;
-  font-size: 15px;
-  color: #ffffff;
-  font-family: 'Fira Code', monospace;
-
-  @media only screen and (max-width: 990px) {
-    width: 100%;
-  }
 
-  @media only screen and (max-width: 769px) {
-    width: 100%;
-  }
+  ${({props}) => `
+    width: ${ props?.styles?.width || "-webkit-fill-available" };
+    border: ${ props?.styles?.border || "none" };   
+    color: ${ props?.styles?.color || "#FFF" };
+    display: ${ props?.styles?.display || "block" };   
+    border-color: ${ props?.styles?.borderColor || "#067c82" };  
+    border-style: ${ props?.styles?.borderStyle || "solid" };  
+    border-width: ${ props?.styles?.borderWidth || "1px" };  
+    border-radius: ${ props?.styles?.borderRadius || "0" };  
+    background: ${ props?.styles?.background || "#08333c" };   
+    padding: ${ props?.styles?.padding || "11px 15px" };    
+    font-size: ${ props?.styles?.fontSize || "15px" };      
+    font-family: ${ props?.styles?.fontFamily || "'Fira Code', monospace" };
+  `}  
 `;
 
-const StyledInput = styled.input`
-  width: 100%;
-  border: none;
-  background-color: inherit;
-  color: #ffffff;
-  //width: auto;
-`;
-
-const HideOnSmallScreen = styled.img`
-  width: 30px;
+const StyledSearchIconWrapper = styled.div`  
   cursor: pointer;
   float: right;
   position: absolute;
-  right: 22px;
-  top: 9px;
-  @media only screen and (max-width: 990px) {
-    display: none;
-  }
+  ${({props}) => `
+    right: ${ props?.styles?.right || "15px" };      
+    top: ${ props?.styles?.top || "12px" };        
+  `}    
 `;
 
+const StyledWrapper = styled.div`
+  position: relative;
+`;
 
 const SearchInput = () => {
   const dispatch = useDispatch();
+  const renderProps = useContext(TokenSearchContext);  
+  const { customSearchInput } = renderProps;
+ 
   const { searchText, networkMap, exchangeMap } = useSelector((state:RootState) => state);
   
-  // Updates the datasets of the results.
+  // Updates the datasets of the results. 
   useEffect(() => {
     // Ensure that the search text fulfills the minimum lenght requirement.
     if (searchText.length >= minStringSearch) {
       dispatch(searchTokenPairs(searchText));
     }
-  }, [dispatch, searchText, networkMap, exchangeMap]);
-
+  }, [dispatch, searchText, networkMap, exchangeMap]); 
+  
 
   const onChangeFilter = (event) => {    
     const value = event.target.value    
     dispatch(setSearchText(value))
   }
 
-  const debounceChangeHandler = useCallback( debounce(onChangeFilter, 350), [searchText])  
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debounceChangeHandler = useCallback( debounce(onChangeFilter, 350), [searchText])
+  
+  const placeholder = customSearchInput?.placeholder ?  customSearchInput?.placeholder : 'Please input token name or address.'
```

Comment:
big picture, it might make sense to use something like _.mergeDeep or some nested ... operator

You could then do something like 
`const configToUse = _.mergeDeep(defaultConfig, optionalSuppliedConfigValues)`

Reply:
It's great.
But all the props are optional, you don't need to define the default value, in the search-war widget, all the component has their own default value, which is wrapped by styled-component.
What do you think of it?

Reply:
no problem.

![image](https://user-images.githubusercontent.com/43188883/163240260-ffb45571-8c58-4283-aa52-8e9e9d73ea81.png)

If the props are optional, just dont include them and the default will be used still.

Reply:
I understand what you mean.
I wrote explicitly in document that we can pass the props or not, no matter what you define the value or not, in terms of the search-bar widget, it has the default value logically.